lib/internal/http2/compat.js

```diff
@@ -181,6 +181,10 @@ class Http2ServerRequest extends Readable {
     return '2.0';
   }
 
+  get socket() {
+    return this.stream.session.socket;
+  }
```

Comment:
Actually, you are right in adding this, as its in the http docs. It does not make much sense to use it for http2, as it will be implementation specific (http, https, http2), and produce different result.

My concern is that, by providing this, we are enabling our users to wreck the internals.
It can stay for now. @jasnell what do you think?